application/org.openjdk.jmc.console.twitter/src/main/java/org/openjdk/jmc/console/twitter/TwitterPlugin.java

```diff
@@ -225,50 +218,91 @@ public Tweeter getAuthorizedTweeter(String username) {
 		return null;
 	}
 
-	public void updateStatus(String username, String message) throws TwitterException {
-		Tweeter tweeter = getAuthorizedTweeter(username);
-		if (tweeter != null) {
-			AccessToken at = createAccessToken(tweeter);
-			Twitter twitter = twitterFactory.getInstance();
-			twitter.setOAuthConsumer(consumerKey, consumerSecret);
-			twitter.setOAuthAccessToken(at);
-			twitter.updateStatus(message);
-			return;
+	private static HttpClient getHttpClient() {
+		InetSocketAddress addr;
+		if (proxyHost != null) {
+			addr = new InetSocketAddress(proxyHost, proxyPort);
 		} else {
-			throw new TwitterException(String.format(
-					"Attempted to update status for an account not defined in the preferences. Please set up the account (%s) in preferences first.",
-					username));
+			addr = null;
+		}
+		HttpClient httpClient = HttpClient.newBuilder().version(HttpClient.Version.HTTP_2)
+				.connectTimeout(Duration.ofSeconds(100)).proxy(ProxySelector.of(addr)).build();
+
+		return httpClient;
+	}
+
+	public void updateStatus(String username, String message) throws Exception {
+		verifyTweeter(username);
+		// form parameters
+		Map<String, String> data = new HashMap<String, String>();
+		data.put("status", message);
+
+		HttpRequest request = HttpRequest.newBuilder().POST(ofFormData(data)).uri(URI.create(UPDATE_STATUS_URL))
+				.setHeader("Authorization", getHeader("POST", UPDATE_STATUS_URL, data))
+				.header("Content-Type", "application/x-www-form-urlencoded").build();
+
+		HttpResponse<String> response = getHttpClient().send(request, HttpResponse.BodyHandlers.ofString());
+		if (response.statusCode() != 200) {
+			JOptionPane.showMessageDialog(null,
+					" Some error occured while updating Twitter status. Please verify your twitter app settings. ");
```

Comment:
Similar to the string usage below, it'd be nice to include the error code along with this message.